app/v1/controllers/requestParsers/validators/SubmitReturnValidator.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators
+
+import v1.controllers.requestParsers.validators.validations.{DateFormatValidation, DecimalValueValidation, JsonFormatValidation, NonDecimalValueValidation, PeriodKeyValidation, UniqueIDValidation, ValueFormatErrorMessages, VrnValidation}
+import v1.models.errors.MtdError
+import v1.models.request.{SubmitReturnRawData, SubmitReturnRequestBody}
+
+class SubmitReturnValidator extends Validator[SubmitReturnRawData] with ValueFormatErrorMessages{
+
+  private val validationSet = List(parameterFormatValidation, bodyValueValidator)
+
+  private def parameterFormatValidation: SubmitReturnRawData => List[List[MtdError]] = (data: SubmitReturnRawData) => {
+    List(
+      VrnValidation.validate(data.vrn),
+      JsonFormatValidation.validate[SubmitReturnRequestBody](data.body.json)
+    )
+  }
+
+  private def bodyValueValidator: SubmitReturnRawData => List[List[MtdError]] = { data =>
+    val requestBodyData = data.body.json.as[SubmitReturnRequestBody]
+
+    List(Validator.flattenErrors(List(
+      PeriodKeyValidation.validate(requestBodyData.periodKey).map(
+        _.copy(paths = Some(Seq(s"/periodKey")))
```

Comment:
Do you need a path here? There's only one periodKey, it's not like `ValueError` which is used for multiple fields.